sdks/java/core/src/test/java/org/apache/beam/sdk/schemas/transforms/JoinTest.java

```diff
@@ -176,6 +182,68 @@ public void testInnerJoinDifferentKeys() {
     pipeline.run();
   }
 
+  @Test
+  @Category(NeedsRunner.class)
+  public void testInnerJoinDifferentKeysNullable2() {
+    List<Row> pc1Rows =
+        Lists.newArrayList(
+            Row.withSchema(CG_SCHEMA_1).addValues("user1", 1, "us").build(),
+            Row.withSchema(CG_SCHEMA_1).addValues("user1", 2, "us").build(),
+            Row.withSchema(CG_SCHEMA_1).addValues("user1", 3, "il").build(),
+            Row.withSchema(CG_SCHEMA_1).addValues("user1", 4, "il").build(),
+            Row.withSchema(CG_SCHEMA_1).addValues("user2", 5, "fr").build(),
+            Row.withSchema(CG_SCHEMA_1).addValues("user2", 6, "fr").build(),
+            Row.withSchema(CG_SCHEMA_1).addValues("user2", 7, "ar").build(),
+            Row.withSchema(CG_SCHEMA_1).addValues("user2", 8, "ar").build(),
+            Row.withSchema(CG_SCHEMA_1).addValues("user3", 8, "ar").build());
+    List<Row> pc2Rows =
+        Lists.newArrayList(
+            Row.withSchema(CG_SCHEMA_2N).addValues("user1", 9, "us").build(),
```

Comment:
Should we test rows with null values as well?

Reply:
Added.